lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceManagerStatus.java

```diff
@@ -147,7 +149,7 @@ private synchronized void onRMDone(final RuntimeStatusEvent runtimeStatusEvent)
     this.driverStatusManager.onComplete();
   }
 
-  private synchronized void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
+  private void onRMRunning(final RuntimeStatusEvent runtimeStatusEvent) {
```

Comment:
onRMFailure calls resourceManagerErrorHandler.onNext() which is synchronized with its own lock, onRMDone calls driverStatusManager.onComplete which also is synchronized in its own class. Both don't use instance variables in ResourceManagerStatus. From this perspective, remove synchronized from other two should be OK. 
But I agree with Mariia we may not want to change until we observe issues.